lib/date_time/date_time.c

```diff
@@ -479,53 +143,42 @@ static void new_date_time_get(void)
 
 		LOG_DBG("Updating date time UTC...");
 
-		err = current_time_check();
+		err = date_time_current_check();
 		if (err == 0) {
 			LOG_DBG("Using previously obtained time");
-			initial_valid_time = true;
 			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_notify_event(DATE_TIME_EVT_TYPE_PREVIOUS);
 			continue;
 		}
 
-		LOG_DBG("Current time not valid");
-
 #if defined(CONFIG_DATE_TIME_MODEM)
 		LOG_DBG("Getting time from cellular network");
+		int64_t date_time_ms_modem = 0;
 
-		err = time_modem_get();
+		err = date_time_modem_get(&date_time_ms_modem);
 		if (err == 0) {
-			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_MODEM;
-			date_time_store(time_aux.date_time_utc);
-			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_store(date_time_ms_modem, DATE_TIME_OBTAINED_MODEM);
 			continue;
 		}
 #endif
 #if defined(CONFIG_DATE_TIME_NTP)
 		LOG_DBG("Getting time from NTP server");
+		int64_t date_time_ms_ntp = 0;
 
-		err = time_NTP_server_get();
+		err = date_time_ntp_get(&date_time_ms_ntp);
 		if (err == 0) {
-			initial_valid_time = true;
-			evt.type = DATE_TIME_OBTAINED_NTP;
-			date_time_store(time_aux.date_time_utc);
-			date_time_schedule_update();
-			date_time_notify_event(&evt);
+			date_time_store(date_time_ms_ntp, DATE_TIME_OBTAINED_NTP);
 			continue;
 		}
 #endif
 		LOG_DBG("Did not get time from any time source");
 
-		evt.type = DATE_TIME_NOT_OBTAINED;
-		date_time_schedule_update();
-		date_time_notify_event(&evt);
+		date_time_notify_event(DATE_TIME_NOT_OBTAINED);
 	}
 }
 
 K_THREAD_DEFINE(time_thread, CONFIG_DATE_TIME_THREAD_STACK_SIZE,
-		new_date_time_get, NULL, NULL, NULL,
+		date_time_new_get, NULL, NULL, NULL,
 		K_LOWEST_APPLICATION_THREAD_PRIO, 0, 0);
 
```

Comment:
Adding a comment here as I cannot add it to `date_time_init()` which starts few lines below.
`date_time_init()` is still a bit of a compilation flag mess which I was kind of hoping to get rid of. However, I'm not sure doing an own function for each of those three function calls in there us much better than that.